### ISSUE TITLE
Loads of images without origin file name

### ISSUE THREAD

**Issue by grololo06** [ecotaxa/ecotaxa_dev#683]
Despite the fact that `img_file_name` is clearly documented as mandatory in the imported TSVs, the corresponding DB column has been historically left as NULL-able.
Consequently, a few bugs and years later:
```
ecotaxa=# create temp table img_no_orig as select * from images where orig_file_name is null;
SELECT 1264703
```
-> 1.26M images without origin

Impacted projects:

ecotaxa=# select 'https://ecotaxa.obs-vlfr.fr/prj/'||projid::varchar, count(1) as cnt from objects where objid in (select objid from img_no_orig) group by projid order by cnt desc;
               ?column?               |  cnt   
--------------------------------------+--------
 https://ecotaxa.obs-vlfr.fr/prj/3081 | 589034
 https://ecotaxa.obs-vlfr.fr/prj/3076 | 359169
 https://ecotaxa.obs-vlfr.fr/prj/3083 | 112316
 https://ecotaxa.obs-vlfr.fr/prj/3077 |  96718
 https://ecotaxa.obs-vlfr.fr/prj/2521 |  69346
 https://ecotaxa.obs-vlfr.fr/prj/3007 |   6245
 https://ecotaxa.obs-vlfr.fr/prj/3006 |   5826
 https://ecotaxa.obs-vlfr.fr/prj/3005 |   4959
 https://ecotaxa.obs-vlfr.fr/prj/3004 |   3065
 https://ecotaxa.obs-vlfr.fr/prj/3010 |   2438
 https://ecotaxa.obs-vlfr.fr/prj/3016 |   2315
 https://ecotaxa.obs-vlfr.fr/prj/3047 |   1904
 https://ecotaxa.obs-vlfr.fr/prj/3018 |   1663
 https://ecotaxa.obs-vlfr.fr/prj/3008 |   1658
 https://ecotaxa.obs-vlfr.fr/prj/3116 |   1626
 https://ecotaxa.obs-vlfr.fr/prj/3069 |   1616
 https://ecotaxa.obs-vlfr.fr/prj/3063 |   1345
 https://ecotaxa.obs-vlfr.fr/prj/3015 |    571
 https://ecotaxa.obs-vlfr.fr/prj/3017 |    552
 https://ecotaxa.obs-vlfr.fr/prj/3064 |    545
 https://ecotaxa.obs-vlfr.fr/prj/3009 |    508
 https://ecotaxa.obs-vlfr.fr/prj/3011 |    465
 https://ecotaxa.obs-vlfr.fr/prj/3040 |    251
 https://ecotaxa.obs-vlfr.fr/prj/3039 |    197
 https://ecotaxa.obs-vlfr.fr/prj/3050 |    195
 https://ecotaxa.obs-vlfr.fr/prj/3056 |    176
(26 lignes)

These cannot be backup-exported reliably as the origin is used to build the .zip when images are required.

Fortunately all images have imgrank = 100, meaning "original image from UVPV6". The fix is just to get the file_name from the image with lowest rank (vignette).

**Comment by jiho** [comment]
How is it possible to import an image without a filename? This is only for UVP6 for which vignettes are generated on the fly upon import?
If so, then the name should indeed be generated (and de-duplicated from the other image for these objects (and solve #679 )

**Comment by grololo06** [comment]
Yep it's just UVP6. 
-  UVP6 data imported before the import code rewrite -> 2 images, one without name. 
-  UVP6 data imported after the import code rewrite -> 2 images with same name.

**Comment by grololo06** [comment]
Data fix on prod DB:
```
ecotaxa=# begin;
BEGIN
Temps : 0,262 ms
ecotaxa=*# update images img
ecotaxa-*# set orig_file_name = (select orig_file_name from images img2 where img2.objid = img.objid and img2.imgrank < img.imgrank)
ecotaxa-*# where orig_file_name is null;
UPDATE 1264703
Durée : 69585,176 ms (01:09,585)
ecotaxa=*# select count(1) from images where orig_file_name is null;
 count 
-------
     0
(1 ligne)

Durée : 3734,998 ms (00:03,735)
ecotaxa=*# commit;
COMMIT
Temps : 0,363 ms
ecotaxa=# 

```

**Comment by grololo06** [comment]
Fixed in 2.5.9.
